ios/browser/api/history/brave_history_api.mm

```diff
@@ -0,0 +1,241 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/history/brave_history_api.h"
+
+#include "base/bind.h"
+#include "base/compiler_specific.h"
+#include "base/containers/adapters.h"
+#include "base/guid.h"
+#include "base/sequenced_task_runner.h"
+#include "base/strings/sys_string_conversions.h"
+#include "base/task/post_task.h"
+#include "base/task/thread_pool.h"
+#include "components/browsing_data/core/browsing_data_utils.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/history/core/browser/history_types.h"
+#include "components/keyed_service/core/service_access_type.h"
+#include "ios/chrome/browser/application_context.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/history/history_service_factory.h"
+#include "ios/chrome/browser/history/web_history_service_factory.h"
+#include "ios/web/public/thread/web_task_traits.h"
+#include "ios/web/public/thread/web_thread.h"
+#include "net/base/mac/url_conversions.h"
+#include "ui/base/page_transition_types.h"
+#include "url/gurl.h"
+
+#include "brave/ios/browser/api/history/brave_history_observer.h"
+#include "brave/ios/browser/api/history/history_service_listener_ios.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+#pragma mark - IOSHistoryNode
+
+@interface IOSHistoryNode () {
+  std::u16string title_;
+  GURL gurl_;
+  base::Time date_added_;
+}
+@end
+
+@implementation IOSHistoryNode
+
+- (instancetype)initWithURL:(NSURL*)url
+                      title:(NSString* _Nullable)title
+                  dateAdded:(NSDate* _Nullable)dateAdded {
+  if ((self = [super init])) {
+    [self setUrl:url];
+
+    if (title) {
+      [self setTitle:title];
+    }
+
+    if (dateAdded) {
+      [self setDateAdded:dateAdded];
+    }
+  }
+
+  return self;
+}
+
+- (void)setUrl:(NSURL*)url {
+  gurl_ = net::GURLWithNSURL(url);
+}
+
+- (NSURL*)url {
+  return net::NSURLWithGURL(gurl_);
+}
+
+- (void)setTitle:(NSString*)title {
+  title_ = base::SysNSStringToUTF16(title);
+}
+
+- (NSString*)title {
+  return base::SysUTF16ToNSString(title_);
+}
+
+- (void)setDateAdded:(NSDate*)dateAdded {
```

Comment:
```#if defined(__OBJC__)
  static Time FromNSDate(NSDate* date);
  NSDate* ToNSDate() const;
#endif``` from `base/time/time.h`

Reply:
Base function is used.  date_added_ = base::Time::FromNSDate(dateAdded);